### PR TITLE
Docs: revamp.

### PR DESCRIPTION
**Pull request by concretevitamin**
Changes
- Ordering of pages
- Removed 2 pages by incorporating them into other pages

Based on traffic in the last few months.

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below): built locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/examples/ports.rst [hunk 1]
```diff
@@ -41,7 +41,7 @@ and look in for the logs for some output like:
 To get the public IP address of the head node of the cluster, run :code:`sky status --ip jupyter`:
 
 .. code-block:: bash
```
Comment (Michaelvll) [inline docs/source/examples/ports.rst:43]:
```suggestion
.. code-block:: console
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -104,9 +104,10 @@ Documentation
    :maxdepth: 1
    :caption: Running Jobs
 
+   examples/spot-jobs
```
Comment (Michaelvll) [inline docs/source/index.rst:107]: For the `gpu-jupyter` above, maybe we can add another section for how to connect vscode to an existing cluster.

File: docs/source/reference/interactive-nodes.rst [hunk 3]
```diff
@@ -86,6 +86,16 @@ Interactive nodes can be stopped, restarted, and terminated, like any other clus
...
+.. note::
+
+    Stopping a cluster does not lose data on the attached disks (billing for the
+    instances will stop while the disks will still be charged).  Those disks
+    will be reattached when restarting the cluster.
+
+    Terminating a cluster will delete all associated resources (all billing
+    stops), and any data on the attached disks will be lost.  Terminated
+    clusters cannot be restarted.
+
```
Comment (Michaelvll) [inline docs/source/reference/interactive-nodes.rst:98]: For the `Getting multiple nodes` below, I would prefer to switch the suggested way to:
`sky launch -c my-cluster --num-nodes 16 --gpus V100:8` to make it easier to create interactive cluster.

**Review by Michaelvll — APPROVED** [review]
Thanks for updating the doc layout @concretevitamin! LGTM.

File: docs/source/reference/interactive-nodes.rst (hunk 3 above)
Comment (concretevitamin) [reply docs/source/reference/interactive-nodes.rst:98]: Done.

File: docs/source/index.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/index.rst:107]: Will leave this for the future!

**Review by concretevitamin — COMMENTED** [review]
Thanks @Michaelvll, addressed and reordered some more. PTAL.

**Review by Michaelvll — APPROVED** [review]

**Pull request merged**